### PR TITLE
Fixes #79 - correctly pass future parser option

### LINKED ISSUE #79: Future Option enabled
When having a version below 4, there is no option to specify that the future option is enabled.
What we have found is that puppet_epp_syntax_check.sh is not parametrized when it is called from pre-commit.  So as the argument is not defined, that script will have the USE_PUPPET_FUTURE_PARSER null  and it will call puppet parse validete without the -parser future  option.

### PR DESCRIPTION
**Pull request by Klaas-**
for the pre-commit hook

should be issue #79 :)

Greetings
Klaas

### CODE REVIEW

File: pre-commit [hunk 1]
```diff
@@ -68,7 +73,7 @@ for changedfile in $files_to_check; do
                 failures=$((failures + 1))
             fi
         elif [[ $(echo "$changedfile" | grep -q '\.*\.pp$'; echo $?) -eq 0 ]]; then
-            ${subhook_root}/puppet_manifest_syntax_check.sh "$changedfile"
+            ${subhook_root}/puppet_manifest_syntax_check.sh "$changedfile" "" "$USE_PUPPET_FUTURE_PARSER"
```
Comment (drwahl) [inline pre-commit]: What's with the empty "" after $changedfile ?

Comment (Klaas-) [inline pre-commit]: $2 is module dir (https://github.com/drwahl/puppet-git-hooks/blob/master/commit_hooks/puppet_manifest_syntax_check.sh)
but its not used with the pre-commit hook; only with the pre-receive hook

File: pre-commit [hunk 2]
```diff
@@ -42,6 +42,11 @@ if [[ -f "$git_root/.git" ]]; then
     fi
 fi
 
+# Only puppet 3.2.1 - 3.8 support "--parser future" option.
+case $(puppet --version) in
+  4*) USE_PUPPET_FUTURE_PARSER="disabled" ;;
```
Comment (krzyzakp) [inline pre-commit]: I would add some information about it to documentation, as now you will need to dig in code to find out, that you can enable it somehow.

Comment (Klaas-) [inline pre-commit]: I did not add that feature, I just extended it to work in pre-commit hook aswell (previously it only worked in pre-receive hook); but I'll happyly add a short description in readme later today :)

Comment (krzyzakp) [inline pre-commit]: True, feature was quite undocumented, but cool that now it's there. Thanks for it!

**Pull request merged**